crates/genesis/src/lib.rs

```diff
@@ -293,15 +293,33 @@ fn deserialize_private_key<'de, D>(deserializer: D) -> Result<Option<B256>, D::E
 where
     D: Deserializer<'de>,
 {
-    let opt_str: Option<String> = Option::deserialize(deserializer)?;
-
-    if let Some(ref s) = opt_str {
-        if s == "0x" {
-            return Ok(None);
+    if deserializer.is_human_readable() {
+        match Option::<String>::deserialize(deserializer)? {
+            Some(ref s) => {
+                if s == "0x" {
+                    return Ok(None);
+                }
+                B256::from_str(s).map(Some).map_err(D::Error::custom)
+            }
+            None => Ok(None),
         }
-        B256::from_str(s).map(Some).map_err(D::Error::custom)
     } else {
-        Ok(None)
+        Option::<B256>::deserialize(deserializer)
+    }
```

Comment:
hmm, why we need to change this here, is this because the non human readable impl for b256 is different?

Reply:
In binary, a B256 will not be serialized as a string, but as 32 bytes. Thus in a roundtrip, the Option<String>::deserialize would fail. It only works for human-readable where it is serialized as a hex string.